server/src/controller/apis/blogs.js

```diff
@@ -1,18 +1,21 @@
 import { Router } from 'express';
 import authInstance from '../../configs/authentication';
-import { authenticateUserBlog, blogExists } from '../../middlewares/blogs';
+import { authenticateUserBlog, createBlogValid } from '../../middlewares/blogs';
 import blogsAPI from '../../services/blogs/api';
+import {
+  saveBlog, fetchBlog, updateBlog, deleteBlog,
+} from '../../services/blogs/service';
 
 const router = Router();
 
 router.get('/health', blogsAPI.health);
 
-router.get('/:id', blogsAPI.readBlog);
+router.get('/:id', fetchBlog, blogsAPI.readBlog);
 
-router.post('/', authInstance.isValidUser(), blogsAPI.createBlog);
+router.post('/', createBlogValid, authInstance.isValidUser(), saveBlog, blogsAPI.createBlog);
```

Comment:
According to this flow: 
- Check if fields passed in the request are valid.
- Check if the user that is going to create a blog is valid or not.
- Create the blog in DB.
- Again create the blog (`blogsAPI.createBlog`).

You can first check whether or not the user is valid or not. Then check if the fields are valid or not and then use any one middlelware or api to create the blog

Reply:
I thought it will unnecessary make a database call to verify the user if the request is invalid. @nimish-gupta what do you think?

Reply:
checking first the validity of request makes more sense, try to reduce as much db calls as possible.

Reply:
:+1: